### ISSUE TITLE
[HOLD] [$1000] mWeb - Contact gets highlighted on long press

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#19766]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Tap on the search icon
2. Long press on a contact just after the page opens

## Expected Result:
It shouldn't get highlighted

## Actual Result:
The contact gets highlighted

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.19.2

**Reproducible in staging?:** n/a

**Reproducible in production?:** n/a

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/93399543/af853fa4-7f8e-402e-810e-2b83a7409b21

**Expensify/Expensify Issue URL:**

**Issue reported by:** @thesahindia

**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1685013467217839

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01458f36110327280c</li>
        <li>Upwork Job ID: 1663571956217389056</li>
        <li>Last Price Increase: 2023-06-06</li>
    </ul>
</details>

**Comment by kushu7** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Contact gets highlighted on long press
### What is the root cause of that problem?
On touch devices, long press sometimes fire hover event. its default behaviour can easily replicate and its not related to only search screen.
 #### bug video

https://github.com/Expensify/App/assets/31707153/2a923150-2619-4f6f-a632-56f191920782

### What changes do you think we should make in order to solve the problem?
We can prevent it by checking `DeviceCapabilities.hasHoverSupport()`
https://github.com/Expensify/App/blob/24381ed092b0ec280083b9454fd6968710e552fd/src/components/Hoverable/index.js#L58-L60
We have two options.
1. We can simply add this condition here.
```js
// should allow hover only when device hasHoverSupport capability.
const isHovered = DeviceCapabilities.hasHoverSupport() && this.state.isHovered;
const child = _.isFunction(this.props.children) ? this.props.children(isHovered) : this.props.children;
```
2. We can return children same as `index.native.js`
```js
render() {
    if(!DeviceCapabilities.hasHoverSupport()) {
    const childrenWithHoverState = _.isFunction(props.children) ? props.children(false) : props.children;
    return <View style={props.containerStyles}>{childrenWithHoverState}</View>;
    }
    .....
}
```
both will solve our problem globally.


### What alternative solutions did you explore? (Optional)
None

**Comment by puneetlath** [comment]
@Santhosh-Sellavel thoughts on the proposal?

**Comment by Santhosh-Sellavel** [comment]
@kushu7's [proposal](https://github.com/Expensify/App/issues/19766#issuecomment-1568799436) Looks good to me! @puneetlath

**Comment by puneetlath** [comment]
Looks to me like @kushu7 offered two different options. @kushu7 which do you prefer and why?

**Comment by kushu7** [comment]
> Looks to me like @kushu7 offered two different options. @kushu7 which do you prefer and why?

I prefer 2nd option, in this we can skip unnecessary function callbacks that are related to hovering.

**Comment by puneetlath** [comment]
@Santhosh-Sellavel  I'll be OOO this week, so if you need Internal help moving this forward during that time please ask in #contributor-plus. Thanks!

**Comment by Santhosh-Sellavel** [comment]
Thanks @amyevans!

**Comment by Santhosh-Sellavel** [comment]
@abekkala  Sorry for the delay. Can you add a hold to the issue title for issue #20585?

**Comment by Santhosh-Sellavel** [comment]
@abekkala we need to add hold to this issue!

**Comment by Santhosh-Sellavel** [comment]
@abekkala we need to add a hold to this issue. Sorry for not being clear, test the PR is blocked by a regression. And it is being worked on the other issue https://github.com/Expensify/App/issues/20585. So we need to hold this until the other PR is merged.

**Comment by abekkala** [comment]
Ah ok - got it!

**Comment by puneetlath** [comment]
@Santhosh-Sellavel it looks like https://github.com/Expensify/App/issues/20585 is done. Does this still need to be held?

**Comment by Santhosh-Sellavel** [comment]
@puneetlath This issue what we discussed yesterday!

**Comment by puneetlath** [comment]
@Santhosh-Sellavel @kushu7 it looks like this was made obsolete by another PR. However, since you were hired and already did the work, we'll still go ahead and pay you out (no bonus) for the effort. 

I'll go ahead and close the PR and I've sent you both hiring offers. Ping me here once you've accepted and I'll pay them out. Thanks for the hard work!

**Comment by puneetlath** [comment]
Alright. @Santhosh-Sellavel will be paid via [this](https://expensify.slack.com/archives/C02NK2DQWUX/p1687451792686509?thread_ts=1686871517.189609&cid=C02NK2DQWUX). @kushu7 has been paid via Upwork. 

Thanks everyone!

**Comment by puneetlath** [comment]
@thesahindia sorry about that. Sent you an offer. 

@Santhosh-Sellavel do you need me to leave the issue open for tracking your payment? Or am I good to close once I pay sahil?

**Comment by puneetlath** [comment]
Ok sounds good. 

@thesahindia please accept the contract when you have a chance.

**Comment by puneetlath** [comment]
Ok, @thesahindia has been paid. So we can close this once @Santhosh-Sellavel has confirmed he's been paid via Expensify.

**Comment by puneetlath** [comment]
@Santhosh-Sellavel did you receive payment yet?

**Comment by puneetlath** [comment]
Everyone has been paid. Closing!
